Prevent adding IPA objects as external members of external groups

abbra opened the pull request:
The purpose of external groups in FreeIPA is to be able to reference
objects only existing in trusted domains. These members get resolved
through SSSD interfaces but there is nothing that prevents SSSD from
resolving any IPA user or group if they have security identifiers
associated.

Enforce a check that a SID returned by SSSD does not belong to IPA
domain and raise a validation error if this is the case. This would
prevent adding IPA users or groups as external members of an external
group.

Fixes: ...
Signed-off-by: Alexander Bokovoy <abokovoy@redhat.com>

flo-renaud commented:
Hi @abbra 
thanks for the PR, the code LGTM. I tried to add an ipa user and an ipa group as external member and it was properly refused:
```
# ipa group-add-member extgrp --external admin@ipa.test
[member user]: 
[member group]: 
[member service]: 
  Group name: extgrp
  External member: S-1-5-21-3005052257-2375221410-442149667-1361,
                   S-1-5-21-3005052257-2375221410-442149667-1380
  Failed members: 
    member user: 
    member group: admin@ipa.test: invalid 'trusted domain object': Object does not belong to a trusted domain
    member service: 
-------------------------
Number of members added 0
-------------------------
# ipa group-add-member extgrp --external admins@ipa.test
[member user]: 
[member group]: 
[member service]: 
  Group name: extgrp
  External member: S-1-5-21-3005052257-2375221410-442149667-1361,
                   S-1-5-21-3005052257-2375221410-442149667-1380
  Failed members: 
    member user: 
    member group: admins@ipa.test: invalid 'trusted domain object': Object does not belong to a trusted domain
    member service: 
-------------------------
Number of members added 0
-------------------------
```
On the other hand, adding AD user/group works well.

abbra commented:
Rebased and removed temporary commit. I also added release notes blurb to the commit message  -- I still need to add support for that to freeipa-tools but we can use this commit to test it, once available.

flo-renaud commented:
Hi @abbra 
thanks for the update, LGTM.

flo-renaud commented:
master:

* 2997a74abcfdb0ad1c0b5356949e557c3b624d3c Prevent adding IPA objects as external members of external groups

flo-renaud commented:
@abbra A manual backport is required for  ipa-4-6 branch
